versioning/spec/versioning.rb

```diff
@@ -0,0 +1,193 @@
+require_relative '../versioning'
+require 'tmpdir'
+
+describe Versioning do
+  def create_commit(file_name)
+    File.write(file_name, 'Dummy content')
+    `git add #{file_name}`
+    `git commit --no-gpg-sign --message Dummy #{file_name}`
+  end
+
+  def create_git_dir_with_tag(tag)
+    `git init`
+    create_commit(tag)
+    `git tag #{tag}`
+  end
+
+  def create_dummy_commit_and_tag(tag)
+    create_commit(tag)
+    `git tag #{tag}`
+  end
+
+  around(:each) do |example|
+    Dir.mktmpdir do |dir|
+      Dir.chdir(dir) do
+        example.run
+      end
+    end
+  end
+
+  describe '.short_git_sha' do
+    it 'returns first 8 git sha1 characters' do
+      Dir.mktmpdir do |dir|
+        Dir.chdir(dir) do
+          create_git_dir_with_tag('v0.0.1')
+          git_sha=`git rev-parse HEAD`
+          expect(Versioning.short_git_sha).to eq(git_sha[0..7])
+        end
+      end
+    end
+  end
+
+  describe '.verify_git!' do
+    # Assuming that git is always there when running rspec
+    context 'when git does not exist' do
+      it 'raises an error' do
+        allow(File).to receive(:executable?).and_return(false)
+        expect {
+          Versioning.verify_git!
+        }.to raise_error(StandardError, /The command `git` does not exist!/)
+      end
+    end
+
+    context 'when current is not a git dir' do
+      it 'raises an error' do
+        Dir.mktmpdir do |dir|
+          Dir.chdir(dir) do
+            expect {
+              Versioning.verify_git!
+            }.to raise_error(StandardError, /The current directory `#{dir}` is not a git tree!/)
```

Comment:
The directory returned by `mktmpdir` may include symlinks that `git` will resolve before generating the error message. E.g. on macOS `/tmp` is just a symlink to `/private/tmp`, but `mktmpdir` returns paths relative to `/tmp`.

```suggestion
            }.to raise_error(StandardError, /The current directory `#{Dir.getwd}` is not a git tree!/)
```